src/inmanta/deploy/timers.py

```diff
@@ -0,0 +1,295 @@
+        def _setup_repair(repair_interval: int) -> tuple[int, str, TaskPriority]:
+            countdown = repair_interval
+            reason = (
+                f"Individual repair triggered for resource {resource} because last "
+                f"repair happened more than {countdown}s ago."
+            )
+            priority = TaskPriority.INTERVAL_REPAIR
+
+            return (countdown, reason, priority)
+
+        def _setup_deploy(deploy_interval: int) -> tuple[int, str, TaskPriority]:
+            countdown = deploy_interval
+            reason = (
+                f"Individual deploy triggered for resource {resource} because last "
+                f"deploy happened more than {countdown}s ago."
+            )
+            priority = TaskPriority.INTERVAL_DEPLOY
+
+            return (countdown, reason, priority)
+
+        # Both periodic repairs and deploys are disabled on a per-resource basis.
+        if self.periodic_repair_interval is None and self.periodic_deploy_interval is None:
+            return
+
+        if not is_dirty:
+            # At the time of the call, the resource is in an assumed good state:
+            # schedule a periodic repair for it if the repair setting is set on a per-resource basis
+            if self.periodic_repair_interval:
+                countdown, reason, priority = _setup_repair(self.periodic_repair_interval)
+            else:
+                return
+
+        else:
+            # At the time of the call, the resource is in an assumed bad state:
+            # schedule a repair or a deploy, whichever has the shortest interval set
+            # on a per-resource basis.
+
+            if self.periodic_repair_interval and self.periodic_deploy_interval:
+                if self.periodic_repair_interval < self.periodic_deploy_interval:
+                    countdown, reason, priority = _setup_repair(self.periodic_repair_interval)
+                else:
+                    countdown, reason, priority = _setup_deploy(self.periodic_deploy_interval)
+            elif self.periodic_repair_interval:
+                countdown, reason, priority = _setup_repair(self.periodic_repair_interval)
+            else:
+                assert self.periodic_deploy_interval is not None  # mypy
+                countdown, reason, priority = _setup_deploy(self.periodic_deploy_interval)
+
+        self.resource_timers[resource].cancel()
+        self.resource_timers[resource].set_timer(
+            countdown=countdown,
+            reason=reason,
+            priority=priority,
+        )
+
+    def remove_resource(self, resource: ResourceIdStr) -> None:
+        """
+        Stop managing the timer for the given resource and
+        cancel the associated re-deployment, if any.
+        """
+        if resource in self.resource_timers:
+            self.resource_timers[resource].cancel()
+            del self.resource_timers[resource]
+
+    def remove_resources(self, resources: Sequence[ResourceIdStr]) -> None:
```

Comment:
Probably more like a collection, since ordering has no meaning, and we'll probably get sets in practice.
```suggestion
    def remove_resources(self, resources: Collection[ResourceIdStr]) -> None:
```